### ISSUE TITLE
Review Testing Strategy

### ISSUE THREAD

**Issue by MackenzieHartung** [ampproject/amp-wp#3249]
## Feature description

As a contributor, I want to have a better testing strategy. 

Our test strategy will be mainly informed by the ideas of Kent C Dodds.

The most important aspects of his thinking is covered in these posts:
- [Write tests. Not too many. Mostly integration.](https://kentcdodds.com/blog/write-tests)
- [Write fewer, longer tests.](https://kentcdodds.com/blog/write-fewer-longer-tests)

Preliminary considerations include the following stack:
- jest for overall test running.
- react-testing-library for setup, manipulation and validation of React components.
- puppeteer for running a limited set of e2e tests.
- axe for accessibility testing.
- storybook for visual unit testing.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* AC1: This ticket is to begin the review and discussion surrounding the existing testing strategy (e.g. e2e tests).
* AC2: Additionally, we should be looking into the automation surrounding accessibility testing, as we examine our testing strategy as a whole.
* AC3: Testing should consider the new editor being developed
 
## Implementation brief

* 

## QA testing instructions

* 

## Demo

* 

## Changelog entry

*
